### PR TITLE
Automated Migration Testing (HAPI-FHIR)

### PR DESCRIPTION
**Pull request by volodymyr-korzh**
Migration test scripts added in scope of this PR are highlighted with **bold** font. 

V5_2_0
- [x] Updated: `MPI_LINK` Added: `GOLDEN_RESOURCE_PID`, `RULE_COUNT`

V5_3_0
- [x] New table: `HFJ_SPIDX_QUANTITY_NRML`
- [x] **Updated: `HFJ_SPIDX_QUANTITY_NRML` table. Added: `PARTITION_ID`, `PARTITION_DATE` columns**
- [x] Updated: `HFJ_RESOURCE` table. Added: `SP_QUANTITY_NRML_PRESENT` column
- [x] **Updated: `HFJ_RES_LINK` table. Added: `TARGET_RESOURCE_VERSION` column**

V5_4_0
- [x] **Updated: `TRM_VALUESET_CONCEPT` table. Added: `INDEX_STATUS`, `SOURCE_DIRECT_PARENT_PIDS`, `SOURCE_PID` columns**
- [x] New Table: `HFJ_BLK_IMPORT_JOB`
- [x] New Table: `HFJ_BLK_IMPORT_JOBFILE`

V5_5_0
- [x] New table: `HFJ_BLK_IMPORT_JOBFILE` **Added: `FILE_DESCRIPTION` column**
- [x] **Updated: `HFJ_RESOURCE` table. Added: `SP_CMPTOKS_PRESENT` column**
- [x] New table: `HFJ_IDX_CMB_TOK_NU` **Added: `PARTITION_ID`, `PARTITION_DATE` columns**
- [x] **Updated: `TRM_VALUESET` Added: `EXPANDED_AT` column**

V5_7_0
- [x] **Updated: `HFJ_RES_VER` table. Added: `RES_TEXT_VC` column**
- [x] **Updated: `MPI_LINK` table. Added: `PARTITION_ID`, `PARTITION_DATE` columns**

V6_0_0
- [x] New table: `BT2_JOB_INSTANCE` **Added: `PARAMS_JSON_LOB`, `TOT_ELAPSED_MILLIS`, `ERROR_MSG` columns**
- [x] New table: `BT2_WORK_CHUNK` **Added: `CHUNK_DATA`, `ERROR_MSG` columns**

V6_1_0
- [x] **Updated: `BT2_JOB_INSTANCE` table. Added: `REPORT` column**

V6_2_0
- [x] **Updated: `BT2_JOB_INSTANCE` table. Added: `FAST_TRACKING`, `UPDATE_TIME` columns**
- [x] **Updated: `BT2_WORK_CHUNK` table. Added: `UPDATE_TIME` column**

V6_3_0
- [x] **Updated: `HFJ_RESOURCE` table. Added: `FHIR_ID` column**

V6_6_0
- [x] **Updated: `HFJ_RESOURCE` table. Added: `SEARCH_URL_PRESENT` column**
- [x] **Updated: `HFJ_TAG_DEF` table. Added: `TAG_VERSION`, `TAG_USER_SELECTED` columns**
- [x] Updated: `HFJ_SEARCH` table. Added: `SEARCH_UUID` column - (already added in 4_0_0)
- [x] Updated: `HFJ_BLK_IMPORT_JOB` table. Added: `JOB_ID` column - (already added in 5_4_0)
- [x] Updated: `HFJ_BLK_EXPORT_JOB` table. Added: `JOB_ID` column - (already added in 4_1_0)
- [x] Updated: `HFJ_RES_SEARCH_URL` table. Added: `RES_SEARCH_URL`, `RES_ID`, `CREATED_TIME` columns
- [x] **Updated: `HFJ_REVINFO` table. Added: `REV`, `REVTSTMP` columns**
- [x] New Table: `MPI_LINK_AUD` **Added: `CREATED`, `UPDATED`, `VECTOR`, `PARTITION_ID`, `PARTITION_DATE` columns**
- [x] **Updated: `HFJ_RESOURCE_MODIFIED` table. Added: `RES_ID`, `RES_VER`, `CREATED_TIME`, `SUMMARY_MESSAGE`, `RESOURCE_TYPE` columns**

V6_8_0
- [x] **Updated: `BT2_JOB_INSTANCE` table. Added: `WARNING_MSG`, `REPORT_VC`, `PARAMS_JSON_VC` columns**
- [x] **Updated: `HFJ_SEARCH` table. Added: `SEARCH_QUERY_STRING_VC`, `SEARCH_PARAM_MAP_BIN` columns**
- [x] **Updated: `HFJ_BLK_IMPORT_JOBFILE` table. Added: `JOB_CONTENTS_VC` column**
- [x] **Updated: `BT2_WORK_CHUNK` table. Added: `WARNING_MSG`, `CHUNK_DATA_VC` columns**
- [x] **Updated: `HFJ_RES_VER` table. Added: `SOURCE_URI`, `REQUEST_ID` columns**

V7_0_0
- [x] **Updated: `BT2_JOB_INSTANCE` table. Added: `USER_NAME`, `CLIENT_ID` columns**

### CODE REVIEW

**Review by nathandoef — APPROVED** [review]
Great work on this Volo. 

I've left a couple comments

File: hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V6_8_0/data/H2_EMBEDDED.sql [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+   'R4',
+   0,
+   '2023-05-01 15:54:05.296',
+   '2023-05-01 15:54:05.296',
+   'JSON',
+   1413284,
+   1678,
+   'Observation',
+   1,
+   '{"resourceType":"Patient","meta":{"profile":["http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"]},"name":[{"family":"Lastname","given":["Firstname"],"prefix":["Mrs"]}],"telecom":[{"system":"phone","value":"1-800-683-1319","use":"home"}],"gender":"female","birthDate":"1960-09-30","address":[{"line":["622 College Street"],"city":"Toronto","state":"Ontario","country":"Canada"}],"generalPractitioner":[{"reference":"Practitioner/P1","display":"Dr S. Freep"}],"managingOrganization":{"reference":"Organization/O1","display":"Bedrock General Hospital"}}',
```
[the remark this answers is not in the document]
Comment (volodymyr-korzh) [reply hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V6_8_0/data/H2_EMBEDDED.sql:32]: Yes, this is test data taken from my local db from the same column

File: hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V5_3_0/data/H2_EMBEDDED.sql [hunk 2]
```diff
@@ -46,23 +46,55 @@ INSERT INTO HFJ_SPIDX_QUANTITY_NRML (
...
+   'value-quantity',
    2,
    'https://unitsofmeasure.org',
    'g',
    -864931808150710347,
    6382255012744790145,
    -1901136387361512731,
-   0.012
+   0.012,
+   NULL,
```
[the remark this answers is not in the document]
Comment (volodymyr-korzh) [reply hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V5_3_0/data/H2_EMBEDDED.sql]: Changed all `null` values to test data

File: hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V5_3_0/data/H2_EMBEDDED.sql [hunk 3]
```diff
@@ -46,23 +46,55 @@ INSERT INTO HFJ_SPIDX_QUANTITY_NRML (
...
+   NULL
```
[the remark this answers is not in the document]
Comment (volodymyr-korzh) [reply hapi-fhir-jpaserver-test-utilities/src/main/resources/migration/releases/V5_3_0/data/H2_EMBEDDED.sql]: Changed all `null` values to test data

**Comment by nathandoef** [comment]
Looks good, thanks for replacing `null` values with real data

**Pull request merged**